engine/core/IgeTexture.ts

```diff
@@ -250,50 +241,35 @@ export class IgeTexture extends IgeAsset {
 	 * @private
 	 */
 	_loadScript (scriptUrl: string) {
-		//ige.textures.onLoadStart(scriptUrl, this);
 
 		if (isClient) {
 			import(scriptUrl)
-				.then((module) => {
-					console.log("Loaded module", module);
+				.then(({image}) => {
```

Comment:
I didn't have anything to test this with at the time, does your game require this functionality and I guess if so, this is working?

Reply:
I have tested this in the old IGE codebase and it works fine, the only caveat is that smart textures now need to include an `export` directive. This seemed like a good trade-off to me but happy to discuss further

Reply:
Here's an example if you need one:

`export var image = {
    render: function (ctx, entity) {
        const centerX = 0;
        const centerY = 0;
        const radius = entity.width();

        ctx.beginPath();
        ctx.arc(centerX, centerY, radius, 0, 2 * Math.PI, false);
        ctx.fillStyle = '#d40000';
        ctx.fill();
        ctx.closePath();
    }
};`

Reply:
Ahh ok. I've been using smart textures differently in my recent usage of the ES6 version. Like yours, they are required to be exported (which I like) but we use them like this:

```typescript
import { squareSmartTexture } from "../assets/textures/smartTextures/square";

...

new IgeTexture("squareSmartTexture", squareSmartTexture);
```

So there's no dynamic loading going on when used this way, but I can see that having both options makes sense.

Also just a heads up, there is a type you can use to define an export as a smart texture e.g.

```typescript
export const squareSmartTexture: IgeSmartTexture = {
	render: function (ctx, entity) {
		ctx.fillStyle = entity.data("fillColor") || "#ffffff";
		ctx.shadowColor = entity.data("glowColor");
		ctx.shadowBlur = entity.data("glowSize");
		ctx.shadowOffsetX = 0;
		ctx.shadowOffsetY = 0;
		ctx.fillRect(-entity._bounds2d.x2, -entity._bounds2d.y2, entity._bounds2d.x, entity._bounds2d.y);

		for (let i = 0; i < entity.data("glowIntensity") || 0; i++) {
			ctx.fillRect(-entity._bounds2d.x2, -entity._bounds2d.y2, entity._bounds2d.x, entity._bounds2d.y);
		}
	}
};
```